assets/js/base/components/text-input/validated.js

```diff
@@ -39,39 +40,51 @@ const ValidatedTextInput = ( {
 	const textInputId = id || 'textinput-' + instanceId;
 	errorId = errorId || textInputId;
 
-	const validateInput = ( errorsHidden = true ) => {
-		if ( inputRef.current.checkValidity() ) {
-			clearValidationError( errorId );
-		} else {
-			setValidationErrors( {
-				[ errorId ]: {
-					message:
-						inputRef.current.validationMessage ||
-						__( 'Invalid value.', 'woo-gutenberg-products-block' ),
-					hidden: errorsHidden,
-				},
-			} );
-		}
-	};
-
-	useEffect( () => {
-		if ( focusOnMount ) {
-			inputRef.current.focus();
-		}
-	}, [ focusOnMount ] );
+	const validateInput = useCallback(
+		( errorsHidden = true ) => {
+			if ( inputRef.current.checkValidity() ) {
+				clearValidationError( errorId );
+			} else {
+				setValidationErrors( {
+					[ errorId ]: {
+						message:
+							inputRef.current.validationMessage ||
+							__(
+								'Invalid value.',
+								'woo-gutenberg-products-block'
+							),
+						hidden: errorsHidden,
+					},
+				} );
+			}
+		},
+		[ clearValidationError, errorId, setValidationErrors ]
+	);
 
 	useEffect( () => {
-		if ( validateOnMount ) {
-			validateInput();
+		if ( isPristine ) {
+			if ( focusOnMount ) {
+				inputRef.current.focus();
+			}
+			if ( validateOnMount ) {
+				validateInput();
+			}
+			setIsPristine( false );
 		}
-	}, [ validateOnMount ] );
+	}, [
+		focusOnMount,
+		isPristine,
+		setIsPristine,
+		validateOnMount,
+		validateInput,
+	] );
```

Comment:
Why did you merge two useEffects? [the general conscious is to separate useEffects ](https://reactjs.org/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns)unless you have a reason?

Reply:
Good point. I undid the merge in 14c4ae8978db24f48d7e4f59412ad75f2cab5015.